Git.hub/PullRequest.cs

```diff
@@ -89,6 +90,23 @@ public List<PullRequestCommit> GetCommits()
             return _client.GetList<PullRequestCommit>(request);
         }
 
+        /// <summary>
+        /// Retrieves diff data for all Commits associated with this pull request.
+        /// </summary>
+        /// <param name="diffUrl">URL for diff data to retrieve</param>
+        /// <returns><see cref="string"/> that contains diff data</returns>
+        public async Task<string> GetDiffData(string diffUrl)
+        {
+            if (string.IsNullOrEmpty(diffUrl))
+            {
+                throw new ArgumentException("Empty diff URL: ", nameof(diffUrl));
+            }
+
+            var request = new RestRequest(diffUrl);
+            var diffData = await _client.ExecuteGetAsync<string>(request);
+            return diffData.Content;
```

Comment:
What happens if a request fails? E.g. 401, 404, 500, etc.

Reply:
It looks like the GIthub plugin gets back text indicating the error message ("Not Found", etc.) and shows an error dialog indicating that it couldn't parse the diff.

I need to debug this further, as it seems even sending the PAT in the headers is not enough for one particular organization.